src/sos-packaging.props

```diff
@@ -9,124 +8,79 @@
+    <SosRequiredBinaries Include="$(ArtifactsBinDir)\Linux-musl.arm64.$(Configuration)\sosdocsunix.txt" TargetRid="linux-musl-arm64" />
 
-    <None Include="$(SOSNETCoreBinaries)" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/linux-musl-arm64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\Linux-musl.arm64.$(Configuration)\libsosplugin.so" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/linux-musl-arm64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\Linux-musl.arm64.$(Configuration)\libsos.so" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/linux-musl-arm64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\Linux-musl.arm64.$(Configuration)\sosdocsunix.txt" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/linux-musl-arm64</PackagePath>
-    </None>
+    <SosRequiredBinaries Include="$(SOSNETCoreBinaries)" TargetRid="osx-x64" />
+    <SosRequiredBinaries Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\libsosplugin.dylib" TargetRid="osx-x64" />
+    <SosRequiredBinaries Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\libsos.dylib" TargetRid="osx-x64" />
+    <SosRequiredBinaries Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\sosdocsunix.txt" TargetRid="osx-x64" />
+  </ItemGroup>
 
-    <None Include="$(SOSNETCoreBinaries)" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/osx-x64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\libsosplugin.dylib" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/osx-x64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\libsos.dylib" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/osx-x64</PackagePath>
-    </None>
-    <None Include="$(ArtifactsBinDir)\OSX.x64.$(Configuration)\sosdocsunix.txt" Pack="true" Visible="false">
-      <PackagePath>$(SOSPackagePathPrefix)/osx-x64</PackagePath>
+  <!-- What and where to pack SOS assets in the final packages. -->
+  <ItemGroup>
+    <None Include="@(SosRequiredBinaries)">
+      <Visible>false</Visible>
+      <Pack>true</Pack>
+      <PackagePath>$(SOSPackagePathPrefix)/%(TargetRid)</PackagePath>
+      <Link Condition="'$(PublishSingleFile)' == 'true'">%(TargetRid)/%(Filename)%(Extension)</Link>
+      <CopyToOutputDirectory Condition="'$(PublishSingleFile)' == 'true' and '%(TargetRid)' == '$(RuntimeIdentifier)'">PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+
+  <!-- This is an ugly workaround needed to be able to do r2r/single file on packages that reference TraceEvent as the package's props inject a double
+  write to the same relative path. This is related to https://github.com/microsoft/perfview/issues/1002 -->
+  <Target Name="RemoveDupeAssemblies"
```

Comment:
Should we have this target in another props file, since this one is specifically for fixing TraceEvent's packaging?  We could move this to the Tools dir prop file and it should still accomplish the same result, right?